### PR TITLE
Update Redis Module API; support for GCC 10 and Ubuntu/Hirsute

### PR DESCRIPTION
**Pull request by rafie**

### CODE REVIEW

File: src/cluster.h [hunk 1]
```diff
@@ -5,11 +5,9 @@
  *      Author: meir
  */
 
-#ifndef SRC_CLUSTER_H_
-#define SRC_CLUSTER_H_
+#pragma once
```
Comment (MeirShpilraien) [inline src/cluster.h:8]: Is it mandotory?

File: src/cluster.h [hunk 2]
```diff
@@ -36,5 +34,7 @@ int Cluster_RefreshCluster(RedisModuleCtx *ctx, RedisModuleString **argv, int ar
 int Cluster_ClusterSet(RedisModuleCtx *ctx, RedisModuleString **argv, int argc);
 int Cluster_ClusterSetFromShard(RedisModuleCtx *ctx, RedisModuleString **argv, int argc);
 
-
-#endif /* SRC_CLUSTER_H_ */
+#define VERIFY_CLUSTER_INITIALIZE(c) \
```
Comment (MeirShpilraien) [inline src/cluster.h:37]: Was it mandatory to move it on this PR?

File: src/cluster.h (hunk 1 above)
Comment (rafie) [reply src/cluster.h:8]: No, it simply looks nicer.

File: src/common.h [hunk 3]
```diff
@@ -22,43 +19,9 @@
 #define MIN(a,b) (((a)<(b))?(a):(b))
 #define MAX(a,b) (((a)>(b))?(a):(b))
 
-extern Gears_dictType* dictTypeHeapIdsPtr;
```
Comment (MeirShpilraien) [inline src/common.h:25]: Was it mandatory to move those in this PR?

File: src/module.h [hunk 4]
```diff
@@ -0,0 +1,15 @@
+
+#pragma once
```
Comment (MeirShpilraien) [inline src/module.h:2]: Why do we need this new file?

File: src/redisgears.h [hunk 5]
```diff
@@ -5,8 +5,15 @@
...
-#define SRC_REDISGEARG_H_
+#pragma once
+
+#ifdef GEARS_API_MAIN
+#define GEARS_API
+#endif
+
+#ifndef GEARS_API
+#define GEARS_API extern
+#endif
```
Comment (MeirShpilraien) [inline src/redisgears.h:16]: If we are already doing this change then why not also add a suffix like the redismodule.h?

**Review by MeirShpilraien — COMMENTED** [review]
Some comments, basically I am not sure it's mandatory to move stuff between files, I believe that if it's not we should not do it in this PR.

**Pull request closed without merging**